src/main/scala/chisel3/simulator/EphemeralSimulator.scala

```diff
@@ -18,12 +18,14 @@ import java.io.File
   */
 object EphemeralSimulator extends PeekPokeAPI {
 
+  private val chiselSim = new ChiselSim {}
+
   def simulate[T <: RawModule](
     module:       => T,
     layerControl: LayerControl.Type = LayerControl.EnableAll
   )(body: (T) => Unit): Unit = {
-    implicit val temporary = HasTestingDirectory.temporary(deleteOnExit = true)
-    DefaultSimulator.simulateRaw(
+    implicit val temporary: HasTestingDirectory = HasTestingDirectory.temporary(deleteOnExit = true)
+    chiselSim.simulateRaw(
```

Comment:
Is the `ChiselSim` object stateful? I assume you're sharing it between multiple runs for a reason (presumably more performant if it isn't stateful) but just want to check that's intentional.

Reply:
No mutability other than that exposed by `TestingDirectory`'s dynamic variable `testName` which is set from a fixture.

Or: other than that, it is definitely immutable as it is just packaging up `SimulatorAPI` (just methods) with `PeekPokeAPI` (methods and classes).

This should be safe.